path/path_test.go

```diff
@@ -28,3 +28,27 @@ func TestPathParsing(t *testing.T) {
 		}
 	}
 }
+
+func TestPathCleaning(t *testing.T) {
+	cases := map[string]string{
+		// .. should strip the preceding path segment
```

Comment:
@whyrusleeping what reservations do you have re `.` and `..` matching unix/posix and the web expectations?